### PR TITLE
Fixed false positive `MISSING_KDOC_ON_FUNCTION` on local function (function inside another function)

### LINKED ISSUE #1773: False positive `MISSING_KDOC_ON_FUNCTION` on local function (function inside another function)
Warning `MISSING_KDOC_ON_FUNCTION` is thrown for functions `printHello()` and `printBye()`:
```
    fun printHelloAndBye() {
        fun printHello() {
            print("Hello")
        }
        printHello()
        val ab = 5
        ab?.let {
            fun printBye() {
                print("Bye")
            }
            printBye()
        }
    }
```

### PR DESCRIPTION
**Pull request by diphtongue**
### What's done:
- Fixed false positive `MISSING_KDOC_ON_FUNCTION` on local function (function inside another function)
- Added warning test

Closes #1773

### CODE REVIEW

File: diktat-rules/src/test/kotlin/com/saveourtool/diktat/ruleset/chapter2/KdocMethodsTest.kt [hunk 1]
```diff
@@ -425,4 +425,28 @@ class KdocMethodsTest : LintTestBase(::KdocMethods) {
             DiktatError(2, 1, ruleId, "${MISSING_KDOC_ON_FUNCTION.warnText()} writeToConsoleEx", true),
         )
     }
+
+    @Test
+    @Tag(WarningNames.MISSING_KDOC_ON_FUNCTION)
+    fun `KDoc shouldn't trigger on local functions`() {
+        lintMethod(
+            """
+                |fun printHelloAndBye() {
```
Comment (akuleshov7) [inline diktat-rules/src/test/kotlin/com/saveourtool/diktat/ruleset/chapter2/KdocMethodsTest.kt:434]: @diphtongue you don't like to write tests, I see 😄 
But pls write some more cases: for example with several nested methods.

And added both scenarios: when Kdoc is presented and when is not

File: diktat-rules/src/main/kotlin/com/saveourtool/diktat/ruleset/rules/chapter1/IdentifierNaming.kt [hunk 2]
```diff
@@ -139,6 +145,36 @@ class IdentifierNaming(configRules: List<RulesConfig>) : DiktatRule(
         return false
     }
 
+    /**
+     * method checks that identifier is correct backing field
+     */
+    private fun ASTNode.isCorrectBackingField(variableName: ASTNode): Boolean {
+        val propertyNodes = this.treeParent.getAllChildrenWithType(KtNodeTypes.PROPERTY)
+        val variableNameCut = variableName.text.drop(1)
+        // check that backing field name is correct
```
Comment (akuleshov7) [inline diktat-rules/src/main/kotlin/com/saveourtool/diktat/ruleset/rules/chapter1/IdentifierNaming.kt:154]: I have seen this code somewhere already... Why backing fields logic was touched here?

File: diktat-rules/src/test/kotlin/com/saveourtool/diktat/ruleset/chapter2/KdocMethodsTest.kt (hunk 1 above)
Comment (diphtongue) [reply diktat-rules/src/test/kotlin/com/saveourtool/diktat/ruleset/chapter2/KdocMethodsTest.kt:434]: Done

File: diktat-rules/src/main/kotlin/com/saveourtool/diktat/ruleset/rules/chapter1/IdentifierNaming.kt (hunk 2 above)
Comment (diphtongue) [reply diktat-rules/src/main/kotlin/com/saveourtool/diktat/ruleset/rules/chapter1/IdentifierNaming.kt]: Deleted

**Review by kgevorkyan — APPROVED** [review]

**Review by DrAlexD — APPROVED** [review]

**Pull request merged**